deploy/contracts/Advertisement.sol

```diff
@@ -0,0 +1,92 @@
+pragma solidity ^0.4.8;
+/**
+ * The AdCampaign contract does this and that...
+ */
+contract Advertisement {
+
+	struct Filters {
+		string countries; 
+		string packageName;
+		uint[] vercodes;
+	}
+
+	struct ValidationRules {
+		bool vercode;
+		bool ipValidation;
+		bool country;
+		uint constipDailyConversions;
+		uint walletDailyConversions;
+	}
+
+	struct Campaign {	
+		bytes8 bid_id;
+		uint price;
+		uint budget;
+		uint startDate;
+		uint endDate;
+		string ipValidator;
+		address  owner;
+		Filters filters;
+	}
+
+	ValidationRules public rules;
+	mapping (bytes => Campaign[]) campaigns;
+
+	event CampaignCreated(string packageName, string countries, 
+							uint[] vercodes, uint price, uint budget,
+							uint startDate, uint endDate);
+
+	function Advertisement () public {
+		rules = ValidationRules(false,true,true,2,1);
+	}
+
+
+	function createCampaign (string packageName, string countries, 
+							uint[] vercodes, uint price, 
+							uint startDate, uint endDate) public payable {
```

Comment:
The contract should have access to the APPC the developer wants to spend in the campaign. Making this function **payable** would be useful if the contract would receive ETH and not APPC, since the **payable** modifier only works with ETH.

Therefore, the flow that would need to happen in order for the contract to have APPC and know about it would be the developer calling the **approve** method in the APPC token contract allowing the contract to spend APPC from the developer's wallet, and then when the **createCampaign** is called, the contract would need to check if it has access to the needed amount of APPC from the user and then transfer them to it via the **transferFrom** method in the APPC token contract.

What do you think?